packages/arb-token-bridge-ui/src/pages/api/cctp/[type].ts

```diff
@@ -154,53 +194,70 @@ export default async function handler(
       l1ChainId === ChainId.Mainnet ? 'cctp-arb-one' : 'cctp-arb-goerli'
     )
 
-    const messagesSentQuery = gql(`{
-        messageSents(
-          where: {
-            sender: "${walletAddress}"
-            ${searchString ? `transactionHash_contains: "${searchString}"` : ''}
-          }
-          orderDirection: "desc"
-          orderBy: "blockTimestamp"
-          first: ${Number(pageSize)}
-          skip: ${Number(pageNumber) * Number(pageSize)}
-        ) {
-          attestationHash
-          blockNumber
-          blockTimestamp
-          id
-          message
-          nonce
-          sender
-          recipient
-          sourceDomain
-          transactionHash
-          amount
-        }
-      }`)
-
-    let messagesSentResult: ApolloQueryResult<{ messageSents: MessageSent[] }>
+    const messagesSentQueryFromWalletAddress = getMessageSents({
+      sender: walletAddress,
+      searchString,
+      pageSize,
+      pageNumber
+    })
+    const messagesSentQueryToWalletAddress = getMessageSents({
+      recipient: walletAddress,
+      searchString,
+      pageSize,
+      pageNumber
+    })
+    let messagesSentToWalletAddressResult: ApolloQueryResult<{
+      messageSents: MessageSent[]
+    }>
+    let messagesSentFromWalletAddressResult: ApolloQueryResult<{
+      messageSents: MessageSent[]
+    }>
     if (type === 'deposits') {
-      messagesSentResult = await l1Subgraph.query({ query: messagesSentQuery })
+      ;[
+        messagesSentToWalletAddressResult,
+        messagesSentFromWalletAddressResult
+      ] = await Promise.all([
+        l1Subgraph.query({
+          query: messagesSentQueryToWalletAddress
+        }),
+        l1Subgraph.query({
+          query: messagesSentQueryFromWalletAddress
+        })
+      ])
     } else {
-      messagesSentResult = await l2Subgraph.query({ query: messagesSentQuery })
+      ;[
+        messagesSentToWalletAddressResult,
+        messagesSentFromWalletAddressResult
+      ] = await Promise.all([
+        l2Subgraph.query({
+          query: messagesSentQueryToWalletAddress
+        }),
+        l2Subgraph.query({
+          query: messagesSentQueryFromWalletAddress
+        })
+      ])
     }
-    const { messageSents } = messagesSentResult.data
-    const messagesSentIds = messageSents.map(messageSent => messageSent.id)
-    const formatedIds = messagesSentIds.map(
-      messageSentId => `"${messageSentId}"`
-    )
+
+    const formattedMessagesSentToIds =
+      messagesSentToWalletAddressResult.data.messageSents.map(
+        messageSent => `"${messageSent.id}"`
+      )
+    const formattedMessagesSentFromIds =
+      messagesSentFromWalletAddressResult.data.messageSents.map(
+        messageSent => `"${messageSent.id}"`
```

Comment:
the id is already a string so no need to wrap it in ``?

Reply:
for graphql. We need to have it as `["id1", "id2", "id3"]`